modules/apps/translation/translation-web/src/main/resources/META-INF/resources/js/translate/Translate.js

```diff
@@ -30,14 +30,49 @@ const ACTION_TYPES = {
 	UPDATE_FIELDS_BULK: 'UPDATE_FIELDS_BULK',
 };
 
+const getInfoFields = (infoFieldSetEntries = []) => {
+	const sourceFields = {};
+	const targetFields = {};
+
+	infoFieldSetEntries.forEach(({fields}) => {
+		fields.forEach(({id, sourceContent, targetContent}) => {
+			sourceFields[id] = sourceContent;
+
+			targetFields[id] = {
+				content: targetContent,
+				message: '',
+				status: '',
+			};
+		});
+	});
+
+	return {
+		sourceFields,
+		targetFields,
+	};
+};
+
+const normalizeFields = (fields = []) =>
+	fields.reduce((acc, field) => {
```

Comment:
Transforms API data to js object.

`[{id1: content1},{id2: content2},{id3: content3}, ...] => {id1: content1, id2: content2, id3: content, ...}`

Reply:
See https://github.com/liferay-lima/liferay-portal/pull/1813 for the additional changes. @boton will cherry-pick the commit and add it here.

Reply:
We removed the necessity of the `prepareFields` function and reduced the complexity of `massageResponseFields` whit these changes (java 5d09197) (js eadad98).

In my opinion, the transforms are cleaner now :D 

![](https://media4.giphy.com/media/IwAZ6dvvvaTtdI8SD5/giphy.gif)

@carloslancha these changes seem fine to you?

Reply:
@carloslancha Are you ok with these changes? 
If looks fine to you, forward it, please 😄

Reply:
Not sure what commits were linked in previous commit but they're not on this pr. Anyway, linking them again since we continued the conversation there: https://github.com/liferay-lima/liferay-portal/commit/eadad98114a6f323d048ef2fb8061c8018de4ab0#